test/process_manager_test.rb

```diff
@@ -91,6 +91,7 @@ class ProcessManagerTest
           end
           pm.stdin.add_data("10\n")
           pm.run!
+          assert_equal '', pm.stderr.to_s
```

Comment:
This doesn't necessarily fix anything, but asserting there were no errors could help when this thing actually fails.